### PR TITLE
[SYCL][E2E] Remove uses of OpenCL primitives in Basic/image e2e tests

### PR DESCRIPTION
**Pull request by ayylol**

### CODE REVIEW

**Comment by againull** [comment]
@ayylol One test is failing in CI, could you please take a look. Overall changes look good to me.

**Comment by aelovikov-intel** [comment]
I think there is a mistake with git branches/merges/rebases or whatever here...

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -5843,10 +5843,15 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-emit-llvm-uselists");
 
     if (IsUsingLTO) {
+      bool IsUsingOffloadNewDriver =
```
Comment (asudarsa) [inline clang/lib/Driver/ToolChains/Clang.cpp:5846]: I think these changes are from a different PR which was recently merged. Can you please sync with SYCL tip? Thanks

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -11120,6 +11125,32 @@ void LinkerWrapper::ConstructJob(Compilation &C, const JobAction &JA,
     for (const auto &A : TranslatorArgs)
       appendOption(OptString, A);
     CmdArgs.push_back(Args.MakeArgString("--llvm-spirv-options=" + OptString));
+
```
Comment (asudarsa) [inline clang/lib/Driver/ToolChains/Clang.cpp:11128]: Same comment as above,

File: sycl/test/check_device_code/hip/atomic/amdgpu_unsafe_atomics.cpp [hunk 3]
```diff
@@ -1,24 +1,49 @@
 // REQUIRES: hip
 // RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx906 %s -S -emit-llvm -o - | FileCheck %s --check-prefixes=CHECK,CHECK-SAFE
 // RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx906 %s -mllvm --amdgpu-oclc-unsafe-int-atomics=true -S -emit-llvm -o - | FileCheck %s --check-prefixes=CHECK,CHECK-UNSAFE
+// RUN: %clangxx -fsycl -fsycl-targets=amd_gpu_gfx90a %s -mllvm --amdgpu-oclc-unsafe-fp-atomics=true  -mllvm --amdgpu-oclc-unsafe-int-atomics=true -S -emit-llvm -o - | FileCheck %s --check-prefixes=CHECK,CHECK-UNSAFE-FP
```
Comment (asudarsa) [inline sycl/test/check_device_code/hip/atomic/amdgpu_unsafe_atomics.cpp:4]: Do we need Compiler Options WG approval for such options? Thanks

File: llvm/lib/Target/AMDGPU/AMDGPUOclcReflect.cpp [hunk 4]
```diff
@@ -39,6 +42,9 @@ static cl::opt<bool>
 static cl::opt<bool> AMDGPUUnsafeIntAtomicsEnable(
     "amdgpu-oclc-unsafe-int-atomics", cl::init(false), cl::Hidden,
     cl::desc("Should unsafe int atomics be chosen. Disabled by default."));
+static cl::opt<bool> AMDGPUUnsafeFPAtomicsEnable(
+    "amdgpu-oclc-unsafe-fp-atomics", cl::init(false), cl::Hidden,
+    cl::desc("Should unsafe fp atomics be chosen. Disabled by default."));
```
Comment (asudarsa) [inline llvm/lib/Target/AMDGPU/AMDGPUOclcReflect.cpp:47]: Please consider rewording: Unsafe fp atomics should be chosen.

**Comment by ayylol** [comment]
Sorry about that, will fix

**Pull request closed without merging**